include/aws/crt/StlAllocator.h

```diff
@@ -51,6 +51,10 @@ namespace Aws
                 return static_cast<RawPointer>(aws_mem_acquire(m_allocator, n * sizeof(T)));
             }
 
+#if _LIBCPP_STD_VER > 20
```

Comment:
do we really need #if here? 
does it really matter if we expose it in all versions of cpp standard?

Reply:
std::allocation_result was added in _LIBCPP_STD_VER > 20